query/stdlib/influxdata/influxdb/rules.go

```diff
@@ -822,37 +819,126 @@ func (rule PushDownGroupAggregateRule) Pattern() plan.Pattern {
 	return plan.OneOf(
 		[]plan.ProcedureKind{
 			universe.CountKind,
+			universe.SumKind,
+			universe.FirstKind,
+			universe.LastKind,
 		},
 		plan.Pat(ReadGroupPhysKind))
 }
 
 func (PushDownGroupAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (plan.Node, bool, error) {
-	// Check the aggregate function spec. Require operation on _value.
-	var aggregateMethod string
-	fnNode := pn
-	switch fnNode.Kind() {
-	case universe.CountKind:
-		if !feature.PushDownGroupAggregateCount().Enabled(ctx) {
-			return pn, false, nil
-		}
-
-		countSpec := fnNode.ProcedureSpec().(*universe.CountProcedureSpec)
-		if len(countSpec.Columns) != 1 || countSpec.Columns[0] != execute.DefaultValueColLabel {
-			return pn, false, nil
-		}
-		aggregateMethod = universe.CountKind
+	group := pn.Predecessors()[0].ProcedureSpec().(*ReadGroupPhysSpec)
+	// Cannot push down multiple aggregates
+	if len(group.AggregateMethod) > 0 {
+		return pn, false, nil
 	}
 
-	groupNode := fnNode.Predecessors()[0]
-	groupSpec := groupNode.ProcedureSpec().(*ReadGroupPhysSpec)
-
-	// Group spec must not already have an aggregate method.
-	if len(groupSpec.AggregateMethod) > 0 {
+	if !canPushGroupedAggregate(ctx, pn) {
 		return pn, false, nil
 	}
 
-	// Rule passes.
-	rewrite := *groupSpec.Copy().(*ReadGroupPhysSpec)
-	rewrite.AggregateMethod = aggregateMethod
-	return plan.CreatePhysicalNode("ReadGroup", &rewrite), true, nil
+	switch pn.Kind() {
+	case universe.CountKind:
+		// ReadGroup() -> count => ReadGroup(count) -> sum
+		root := plan.CreatePhysicalNode("sum", &universe.SumProcedureSpec{
+			AggregateConfig: execute.DefaultAggregateConfig,
+		})
+
+		leaf := plan.CreatePhysicalNode("ReadGroupAggregate", &ReadGroupPhysSpec{
+			ReadRangePhysSpec: group.ReadRangePhysSpec,
+			GroupMode:         group.GroupMode,
+			GroupKeys:         group.GroupKeys,
+			AggregateMethod:   universe.CountKind,
+		})
+
+		root.AddPredecessors(leaf)
+		leaf.AddSuccessors(root)
+		return root, true, nil
+	case universe.SumKind:
+		// ReadGroup() -> sum => ReadGroup(sum) -> sum
+		root := plan.CreatePhysicalNode("sum", &universe.SumProcedureSpec{
+			AggregateConfig: execute.DefaultAggregateConfig,
+		})
+
+		leaf := plan.CreatePhysicalNode("ReadGroupAggregate", &ReadGroupPhysSpec{
+			ReadRangePhysSpec: group.ReadRangePhysSpec,
+			GroupMode:         group.GroupMode,
+			GroupKeys:         group.GroupKeys,
+			AggregateMethod:   universe.SumKind,
+		})
+
+		root.AddPredecessors(leaf)
+		leaf.AddSuccessors(root)
+		return root, true, nil
+	case universe.FirstKind:
+		// ReadGroup() -> first => ReadGroup(first) -> min
+		root := plan.CreatePhysicalNode("min", &universe.MinProcedureSpec{
+			SelectorConfig: execute.DefaultSelectorConfig,
```

Comment:
This should be `min(_time)` so I don't think you can use the default selector config here. In Flux `first` just means the first row of a table, but users expect that to be the oldest point, and `ReadGroup` won't necessarily return data in time order.

Reply:
Thanks for catching this. I must have been on autopilot.

Reply:
Good point. I'm fine with you leaving it as-is.

Reply:
Oops, I meant to reply to a different comment.